File: typescript/Behavioral/13-ChainOfResponsibility/Model/ApprovalRequest.ts
export class ApprovalRequest {
  constructor(
    readonly jobId: string,
    readonly title: string,
    readonly cost: number,
    readonly category: string,
  ) {}

  toString(): string {
    return `Job #${this.jobId} '${this.title}' [${this.category}] — N$${this.cost.toFixed(2)}`;
  }
}

File: csharp/Behavioral/13-ChainOfResponsibility/Model/ApprovalRequest.cs
public class ApprovalRequest
{
    public string JobId { get; }
    public string Title { get; }
    public decimal Cost { get; }
    public string Category { get; }

    public ApprovalRequest(string jobId, string title, decimal cost, string category)
    {
        JobId = jobId;
        Title = title;
        Cost = cost;
        Category = category;
    }

    public override string ToString() =>
        $"Job #{JobId} '{Title}' [{Category}] — N${Cost:N2}";
}
